app: http: Handle HTTP chunked transfer encoding (#290)
nrfconnect/ncs-serial-modem: doc/app/at_httpc.rst

* The ``<length>`` parameter is an integer.
  It contains the number of body bytes in this chunk.

.. note::

>    When the server uses ``Transfer-Encoding: chunked``, the body bytes delivered via ``#XHTTPCDATA`` include the raw chunked framing: each chunk is preceded by its size in hexadecimal followed by ``\r\n``, and followed by ``\r\n``.

Reviewer: ```suggestion
   When the server uses ``Transfer-Encoding: chunked``, the body bytes delivered using ``#XHTTPCDATA`` include the raw chunked framing.
   Each chunk is preceded by its size in hexadecimal followed by ``\r\n``, and followed by ``\r\n``.
```

>    The final zero-length chunk ``0\r\n\r\n`` is also forwarded.  The host is responsible for stripping this framing to recover the original body content.

Reviewer: ```suggestion
   The final zero-length chunk ``0\r\n\r\n`` is also forwarded. 
   The host strips this framing to recover the original body content.
```

>    The ``<total_bytes>`` field in ``#XHTTPCSTAT`` reflects the raw wire byte count including chunked framing, not the decoded body length.

Reviewer: ```suggestion
   The ``<total_bytes>`` field in ``#XHTTPCSTAT`` reflects the raw wire byte count, including chunked framing, not the decoded body length.
```

* The ``<length>`` parameter is an integer.
  It contains the number of body bytes in this chunk.

.. note::

   When the server uses ``Transfer-Encoding: chunked``, the body bytes delivered via ``#XHTTPCDATA`` include the raw chunked framing.

>    Each chunk is preceded by its size in hexadecimal followed by ``\r\n``, and followed by ``\r\n``.

Reviewer 2: Is there `\r\n` twice or is this a leftover in the doc?

Author: Updated the sentence. It was ambiguous.
